src/engine/client/audio.cpp

```diff
@@ -328,3 +317,52 @@ void AudioManager::PlayGlobalSound(const std::string &path) {
     global_source_.reset(CreateSource(path));
     global_source_->StartPlaying();
 }
+
+void AudioManager::SilenceSources() {
+    for(auto sound : sources_) {
+        sound->StopPlaying();
+    }
+
+    global_source_->StopPlaying();
+}
+
+void AudioManager::FreeSources() {
```

Comment:
I think this function wants a warning along the lines of "Will invalidate ALL references to AudioSource objects. Only use in contexts where this is safe."